src/cycax/cycad/assembly.py

```diff
@@ -0,0 +1,236 @@
+                "part_no": item.part_no,
+                "rotmax": item.rotmax,
+                "moves": item.moves,
+                "rotate": item.rotate,
+                "colour": item.colour,
+            }
+            dict_out.append(dict_part)
+        return dict_out
+    
+
+    def rotateFreezeTop(self, part):
+        """
+        This method will hold the front and the left while holding the top where it currently is.
+        Args:
+            part(CycadPart): This is the part that will be rotated. 
+        """
+
+        part.rotate[2] = part.rotate[2] + 90
+        part.x_max, part.y_max = part.y_max, part.x_max
+        part.x_min, part.y_min = part.y_min, part.x_min
+        part.make_bounding_box()
+        part.rotmax = [
+            part.x_max - part.x_min,
+            part.y_max - part.y_min,
+            part.z_max - part.z_min,
+        ]
+
+    def rotateFreezeLeft(self, part):
+        """
+        This method will rotate the top and front while holding the left where it currently is.
+        Args:
+            part(CycadPart): This is the part that will be rotated. 
+        """
+
+        part.rotate[0] = part.rotate[0] + 90
+        part.y_max, part.z_max = part.z_max, part.y_max
+        part.y_min, part.z_min = part.z_min, part.y_min
+        part.make_bounding_box()
+        part.rotmax = [
+            part.x_max - part.x_min,
+            part.y_max - part.y_min,
+            part.z_max - part.z_min,
+        ]
+
+    def rotateFreezeFront(self, part):
+        """
+        This method will rotate the left and top while holding the front where it currently is.
+        Args:
+            part(CycadPart): This is the part that will be rotated. 
+        """
+
+        part.rotate[1] = part.rotate[1] + 90
+        part.x_max, part.z_max = part.z_max, part.x_max
+        part.x_min, part.z_min = part.z_min, part.x_min
+        part.make_bounding_box()
+        part.rotmax = [
+            part.x_max - part.x_min,
+            part.y_max - part.y_min,
+            part.z_max - part.z_min,
+        ]
+
+    def level(self, part1, side1: str, part2, side2: str):
+        """
+        level takes the plane of part 2 specified and moves part 1 so that its specified side has a plane equal to part 2.
+        part1 "FRONT" part2 "BACK" will gve part 1 and part 2 a front and back which are on the same plane. It moves part1.
+        Args:
+            part1(CycadPart):This is the part that will be moved to match the plane of the other part.
+            side1: This is the side of part1 which will be moved around to match the plane of part2
+            part2(CycadPart): This is the part that will used to reference the moving of part1.
+            side2: This is the side which will dictate the plane used to as a reference to move part1.    
+        """
+        part2.make_bounding_box()
+        part1.make_bounding_box()
+        to_here = part2.bounding_box[side2]
+        if side1 == BOTTOM:
```

Comment:
please use `if/elif/else raise` pattern. It more economical and reads nicer. Do you know since version 3.10 Python has match statements? These might be nicer.